src/ProductBundle/Controller/CollectionController.php

```diff
@@ -21,46 +25,98 @@
 class CollectionController extends Controller
 {
     /**
-     * List the main collections.
-     *
-     * @return Response
+     * @var SeoPage
      */
-    public function indexAction(Request $request)
-    {
-        $pager = $this->get('sonata.classification.manager.collection')
-            ->getRootCollectionsPager($request->get('page'));
+    private $sonataSeoPage;
 
-        return $this->render('@SonataProduct/Collection/index.html.twig', [
-            'pager' => $pager,
-        ]);
+    /**
+     * @var CurrencyDetectorInterface
+     */
+    private $currencyDetector;
+
+    /**
+     * @var CollectionManagerInterface
+     */
+    private $collectionManagerInterface;
+
+    /**
+     * @var ProductSetManager
+     */
+    private $productSetManager;
+
+    public function __construct(SeoPage $sonataSeoPage = null, CurrencyDetectorInterface $currencyDetector = null, CollectionManagerInterface $collectionManagerInterface = null, ProductSetManager $productSetManager = null)
```

Comment:
Please break this

Reply:
Sorry what do you mean?

Reply:
```suggestion
    public function __construct(
        SeoPage $sonataSeoPage = null, 
        CurrencyDetectorInterface $currencyDetector = null, 
        CollectionManagerInterface $collectionManagerInterface = null, 
        ProductSetManager $productSetManager = null
    )
```
It is more readeable